### PR TITLE
Introduction to Containers and Docker

### LINKED ISSUE #2154: Introduction to Containers and Docker.
### Introduction paragraph (2-3 paragraphs):
Container is a standard unit of software development in which software service are packaged together so that the application runs quickly across all computing environment. The containerized application can be tested as a unit and deployed as a container image instance to the target operating system.

Docker container image is a lightweight standalone executable package of software that includes system tools, system libraries, code and settings used to run an application. Developers can use development environments on Windows, Linux, or macOS on a computer, the developer runs a Docker host where Docker images are deployed.
### Key takeaways: 
   1. Brief introduction to Containers and Dockets.
   2. Container Terminology
   3. Introduction to Docker/Docker Engine.
   4. Reasons why we use Dockers.
   5. Docker Windows System Requirements and installation.

### PR DESCRIPTION
**Pull request by Evanslodoctor**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

This fixes #2154

### CODE REVIEW

**Comment by hectorkambow** [comment]
@Evanslodoctor 
Good afternoon and thank you for submitting your article for publishing.
Your article has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. 🚀

Thank you.

File: content/articles/Introduction-to-Containers-and-Docker/index.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+5. **Repository**: A Docker repository is a collection of Docker images that all have the same name but differ in tags or versions.
+6. **Registry**: The library is assistance liable for facilitating and circulating pictures.
+7. **Docker Engine**: Docker Engine, is a piece of software that allows us to manage the lifecycle of containers, describe how they will be set up, what applications/software/services they will run within them, their networking and storage needs, and if necessary, how to quickly destroy and restart them.
+8. **Docker Compose**: Docker Compose is an order line instrument for building and running applications made out of numerous Docker containers.
+9. **Orchestration and cluster management**: Orchestration is a tool that monitors and manages the system. In huge compartment arrangements, tooling is crucial for screen and oversee pictures, containers, and hosts through a Command line interface (CLI) or a Graphical user interface(GUI). 
+10. **Volumes**:  Volumes are documents that are mounted on the host and can be imported to different compartments. In addition, it is where all progressions are made to the host filesystem.
```
Comment (Bayler) [inline content/articles/Introduction-to-Containers-and-Docker/index.md:26]: How you've defined the following terminologies is not clear:

1. Container
2. Docker image: ....did you mean, "...building a Docker image?"
3. Registry
4. Docker Compose: What do you mean by order line instrument?
5. Volumes: What do you mean by progressions?

These sentences are either not clear or difficult to follow. Put yourself in the context of a beginner whose reading your article. Either re-write or rephrase the sentences.

File: content/articles/Introduction-to-Containers-and-Docker/index.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+2. **Containers in Docker are freely linked.**: A container is a self-contained entity with its resource quota, networking configuration, and other features that make it easier for system administrators to replace or update one container without impacting the others.
+
+3. **Highly secure**: Docker's Highly Secure feature ensures that containers are entirely isolated from outside processes.
+
+4. **Docker containers are small and portable**: Docker containers are lighter than virtual machines since they just establish separate userspace and use the underlying OS Kernel of the system on which they are run.
+
+5. **It is portable**: When we finish building an application on our local workstation and need to deploy it to production, we run into several problems with environment configuration. However, if you use Docker, you can describe the container setup procedures, and Docker will ensure that the environment is set up the same way every time you run it.
+
+### Docker Windows framework necessities and establishments
+
```
Comment (Bayler) [inline content/articles/Introduction-to-Containers-and-Docker/index.md:60]: A title like "Docker Desktop for Windows requirements and installation" would be more appropriate

File: content/articles/Introduction-to-Containers-and-Docker/index.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+1. A Windows 10 64-bit OS. Also, the `Virtualization and Containers` Windows feature should be enabled.
+
+2. A minimum of 4GB system RAM is recommended. 
+
+> When Hyper-V is activated on Windows, any designed VirtualBox on your framework will quit working. 
+
+#### Docker Installation on Windows Step-By-Step. 
+
+1. Download the docker work area for windows executable record. To download the official file, [click here](https://docs.docker.com/docker-for-windows/install).
```
Comment (Bayler) [inline content/articles/Introduction-to-Containers-and-Docker/index.md:75]: Rephrase this sentence to "Download the `Docker Desktop for Windows` executable record"

File: content/articles/Introduction-to-Containers-and-Docker/index.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+3. Accept the Terms and Agreement to allow the installation cycle to finish.
+
+4. After completion of the installation process, click Finish and launch the Docker Desktop application.
+
+You have successfully launched a Docker Desktop application!
+
+### Conclusion
+
+We learned about containers, containerization, docker or docker-engine, docker client-server architecture, and why Docker is so popular in the software business these days in this article. I urge the reader to dig deeper into docker to be a better person in software development.
```
Comment (Bayler) [inline content/articles/Introduction-to-Containers-and-Docker/index.md:87]: Any references you can add to your article?

File: content/articles/Introduction-to-Containers-and-Docker/index.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
+To comprehend what `Docker` is and why it is used, it is necessary to first understand what `containers` are and what problems they solve. Containers are completely isolated environments built on top of an existing operating system to provide a `virtual barrier` between the app running inside and the rest of the world. Docker, on the other hand, is a piece of software that allows us to manage the lifetime of these containers.
+
+### Table of content
```
Comment (Bayler) [inline content/articles/Introduction-to-Containers-and-Docker/index.md:3]: Any prerequisites needed for one to understand the article? Consider your reader being someone who is reading about Docker for the first time.

**Comment by Bayler** [comment]
Hi @Evanslodoctor I forgot to welcome you to our Engineering program. I have had a good read. You're very knowledgeable on the subject. Don't worry about making mistakes as this is your first article. You'll learn to better your writing skills with time. Have a look at the minor edits I've done on your article to avoid making similar mistakes in the future. In case of anything you don't understand, feel free to ping me on the Slack channel. 

Please work on the requested revisions then we can proceed for final review.

**Comment by hectorkambow** [comment]
@Evanslodoctor 

👋 Good afternoon - and welcome to the EngEd program.
I'll be running a final review on your PR when ready.
Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on plagiarism and what is considered as such.

[Introduction to Containers and Docker #2852.pdf](https://github.com/section-engineering-education/engineering-education/files/6803505/Introduction.to.Containers.and.Docker.2852.pdf)



cc @Bayler

**Comment by Evanslodoctor** [comment]
hello @Bayler 
done correcting requested changes. please check
thank you

**Comment by Bayler** [comment]
> hello @Bayler
> done correcting requested changes. please check
> thank you

@Evanslodoctor did you also make changes concerning plagiarism?

**Comment by Evanslodoctor** [comment]
Hello @Bayler 
plagiarism corrected.

**Comment by Evanslodoctor** [comment]
hello @Bayler wished to inquire the progress my article. Thank you

**Comment by Bayler** [comment]
> hello @Bayler wished to inquire the progress my article. Thank you

I'm done reviewing it @Evanslodoctor. However, an article cannot be added the "Ready for final review" label until Plagiarism check has been performed on your article. Once that is done, it'll be ready for final review.

cc: @hectorkambow

**Comment by ninjaginja** [comment]
plagiarism check completed - all good
[2120420-22996517a603422142fc-1626802383624.pdf](https://github.com/section-engineering-education/engineering-education/files/6850194/2120420-22996517a603422142fc-1626802383624.pdf)

**Comment by Bayler** [comment]
@Evanslodoctor Please adjust the size of your hero image. It's too large. The acceptable dimension for a hero image should be 600px by 400px. It's very clear on the article checklist section. Please avoid making such mistakes in your future article submissions.

**Comment by Evanslodoctor** [comment]
hello @Bayler I have added the requested hero image with 600 by 400px. 
Please check. Thank you

**Comment by louisefindlay23** [comment]
Hi, @Evanslodoctor. From previewing your article, your hero image has a black border on the left and the image (particularly the text) looks a bit squashed. Could you please adjust this?

**Comment by louisefindlay23** [comment]
Thank you for submitting your article.

After some careful consideration, and after reviewing your article for publication, it struck us that this topic is already well covered in our EngEd library and it would be repeating some previous articles.

Due to this we will be closing this PR to avoid topic overlap. We encourage you to submit a more in depth topic or one that builds upon what we already have. Thank you.

cc - @Bayler.

**Comment by Evanslodoctor** [comment]
@louisefindlay23 and @hectorkambow that's not fair.
Closing a PR after topic approval, passing all reviews then entering final review then all over a sudden its all in vain. All time committed while writing just for it to be closed after all that work is done is pains.  Thank you

**Comment by hectorkambow** [comment]
@Evanslodoctor 

Our sincerest apologies and any inconvenience - and we truly encourage you to submit a more in depth version on this topic for you to publish. 

We do our best to avoid overlap in topics (and catch in in the beginning of the topic suggestion phase) but with so many articles being contributed we can only catch so much - that is why we try to make it very clear to student to check that their topic does not overlap with existing articles/topics as well.

We to need to ensure the content we are producing and curating is adding value to the larger community of developers.
It is also not fair to have contributors repeat information and get credit for it when it adds no extra value to the community.

**Hope that helps shed some light on the situation - of course every situation is different and we try to handle them case by case.**

**Pull request closed without merging**